### PR TITLE
Fixed all the Sonar Cloud issues in the Test project

### PR DESCRIPTION
**Pull request by lmalenfant**
Fixed all the code smells in the Test project, there were about 31 of them in total.

### CODE REVIEW

File: src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs:1]: I had a question about the change on line 94: when you modify an "out" parameter from type IDetector to var, does it not loosen the constraint on the parameter?

Comment (lmalenfant) [reply src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs:1]: I don't believe so because it will only suggest using var if there is no ambiguity. This out value is not used in the unit test though so we should probably discard its value with and underscore.

File: src/Vts.Test/MonteCarlo/Tissues/InfiniteCylinderTissueRegionTests.cs
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Tissues/InfiniteCylinderTissueRegionTests.cs:1]: Why the change on line 61-62 from Double.PositiveInfinity to double.PositiveInfinity?  Aren't they equivalent?

File: src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs [hunk 1]
```diff
@@ -139,13 +139,13 @@ public void Demonstrate_user_defined_detector_usage()
             };
             var simInput = new SimulationInput
                 {
-                    DetectorInputs = new[] { detectorInput },
+                    DetectorInputs = new IDetectorInput[] { detectorInput },
```
Comment (lmalenfant) [inline src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs:142]: I changed this here but it is suggesting this change elsewhere in this file, so I think I need to change it in a few more places. This is the reason behind the change:
https://www.jetbrains.com/help/resharper/2023.2/CoVariantArrayConversion.html

**Review by hayakawa16 — APPROVED** [review]
Thanks for your help!  They look good.

File: src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs
Comment (hayakawa16) [reply src/Vts.Test/MonteCarlo/Factories/DetectorFactoryTests.cs:1]: I like fix.
